module/skill/GetUp/src/GetUp.cpp

```diff
@@ -60,67 +61,73 @@ namespace module::skill {
             cfg.getup_upside_down = config["scripts"]["getup_upside_down"].as<std::vector<std::string>>();
         });
 
-        on<Provide<GetUpTask>, Needs<BodySequence>, With<Sensors>>().then(
-            [this](const RunReason& run_reason, const Sensors& sensors) {
+        on<Provide<GetUpTask>, Uses<BodySequence>, With<Sensors>>().then(
+            [this](const RunReason& run_reason, const Uses<BodySequence>& body, const Sensors& sensors) {
                 switch (run_reason) {
```

Comment:
I feel like this might look nicer if it is something like

```
if (run_reason == RunReason::NEW_TASK) {
  // Wait
}
else if (run_reason == RunReason::SUBTASK_DONE && !body.done) {
  // side check and script emit
  }
else if (run_reason == RunReason::SUBTASK_DONE) {
  // the Done emit etc
}
else {
  Continue
}
  ```
  
  The more nesting, the higher the cognitive load for someone reading